### ISSUE TITLE
Is PBN command required?

### ISSUE THREAD

**Issue by mikhail-barg** [Microsoft/CNTK#1035]
I am investigating Batch Normalization and for now it looks like a huge improvement to the learning speed, but at the same time inplace validation (using `cvReader`) shows worse errors. 

E.g. without Batch Normalization the process would get to verification error=0.76 at epoch 30 (starting from err=1.1). With Batch Normalization the process gets to verification error of 0.78 already at 3rd epoch, but then verification error starts to grow, while training error still decreases (looks like a classical overfit).

We are trying changing different parameters but with no luck for now. So I stumbled upon this page: 
https://github.com/Microsoft/CNTK/wiki/Post-Batch-Normalization-Statistics
It refers to the [PBN Command](https://github.com/Microsoft/CNTK/wiki/Top-level-commands#pbn-command). And as far as I get, this command is required to make the Batch Normalization to work properly. Still there's no examples or other code in the repo using it. Do I actually need this and how should I use it considering that it's a separate command from Test or Train?

Also, does using `cvReader` during the Train command give same result as using CV or Test command after training, in regard to Batch Normalization? I'm worried about inference statistics being different.

And one more thing. Could you please check if the following network structure makes sense at all? 
```
model = Sequential(
        DenseLayer{512, activation=Sigmoid} :
        BatchNormalizationLayer { } :
        DenseLayer{256, activation=Sigmoid} :
        BatchNormalizationLayer { } :
        DenseLayer{parameterDim, activation=ReLU}
)
```
I'm worried about having sigmoid non-linearities before BetchNorm.

**Comment by yuxiaoguo** [comment]
For now, only ResNet for ImageNet uses the PBN features, and example link: [CNTK/Examples/Image/Classification/ResNet/BrainScript/ResNet50_ImageNet1K.cntk.](https://github.com/Microsoft/CNTK/blob/master/Examples/Image/Classification/ResNet/BrainScript/ResNet50_ImageNet1K.cntk). The configuration of reader should be same with that in training.

**Comment by DemjanGrubic** [comment]
@yuxiaoguo ResNet for ImageNet uses PBN only after training and before test.
Is it possible to configure .cntk file in such a way that cvReader uses PBN after every epoch?
I would like to have validation accuracy per epoch graph.

One way of doing that would be writing my own script that uses models from every epoch that I got from training and doing commands PBN and Test.
Is there an easier way?

**Comment by cha-zhang** [comment]
According to @yuxiaoguo, PBN don’t have significant improvement, comparing with current exponential running average BN in training. We tested and on ResNet it will boost only 0.01%, while it will drop about around 0.02% in GoogLeNet.

PBN is used in two scenes:
1.	In some cases, people want to know the intrinsic testing curve in training. Meanwhile, the running average BN is instable while training in early stage. So, they enforce evaluating the model with the last batch BN statistics in training (BNTimeConst=0). There is how the test curves in ResNet papers draw. After training, the final test error will be around 26.5%. Since the training is over, PBN is imported to statistics BN mean and variance. After PBN, the result will be 24.7%, reporting by the paper and our early experiments in CNTK. 
2.	Almost same, if someone wants to evaluate the model in training while the gradient still changed frequently and violently, the exponential running average will fail to reflect the real testing error. PBN is also needed here.

You should carefully check the BNTimeConst value in your script. If the value is smaller than 0, meaning your statistics with normal running average/last batch statistics, use PBN will promote the final test error. If your value is bigger than 0, meaning you statistics with exponential running average BN, the result may be almost same.
